### DISCUSSION TITLE
Suggestion: Migra fork

### DISCUSSION THREAD

**Discussion by anarkrypto** [supabase/supabase#19641] [General]
[Migra](https://github.com/djrobstep/migra) is the default tool chosen by Supabase to show the database schema diff, enabling a simple and smart migration using the CLI:

```bash
supabase db diff --local
```
The problem is: **Migra is almost dead!**

The Migra project has not received any updates for a long time and there are currently dozens of serious issues and feature requests in Migra and it's dependencies, some of which have been fixed voluntarily by developers through PRs, but neglected by the only Migra maintainer.

An example is https://github.com/djrobstep/schemainspect/pull/67 PR, which was created almost 3 years ago, but the Migra author didn't merge it and neither provided an explanation. 

Without this functionality, the Supabase CLI is unable to migrate view, functions, and column privileges (see https://github.com/supabase/cli/issues/1716), hindering the resolution of security permission issues when opting for the end-to-end model (user from the browser to the database), which is strongly encouraged by Supabase. 

While we could use third-party tools to work around these limitations, it significantly diminishes the purpose of using Supabase, as its value lies not only in the use of Postgres but also in its entire ecosystem, including the Dashboard, API, SDK, and CLI.

So, it doesn't seem to make sense that a great and popular project like Supabase would depend on a discontinued dependency like Migra.

I believe these are the options:

1. Replace Migra with some other similar project
2. Fork the Migra

Assuming that there is no other good current alternative besides migra, the easiest way is to fork and continue the project.

**Comment by MiroslavPetrik** [comment]
[The `migra` command is now `results dbdiff`](https://github.com/djrobstep/results?tab=readme-ov-file#the-migra-command-is-now-results-dbdiff)
